File: licentainangular.client/src/app/models/Produs.ts
import { User } from './User';

export interface Produs {
  IdProdus: number;
  Nume: string;
  Descriere?: string; // Optional
  EsteSpray: boolean;
  CodCuloare?: string; // Optional
  Imagine: Uint8Array | null; // Optional
  Pret: number;
  Valabil: boolean;
  Categorie: string;
  IdUser: number | null; 
  User: User | null;  // Mark User as nullable (null allowed)
}

File: LicentaInAngular.Server/DataLayer/Models/Produs.cs
using LicentaInAngular.Server.DataLayer.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LicentaInAngular.Server.Models
{
    public class Produs
    {
        [Key] 
        public int IdProdus { get; set; }

        [Required]  
        
        public string Nume { get; set; }

        public string? Descriere { get; set; } 

        [Required]
        public bool EsteSpray { get; set; } 
        [Required]
        public bool Valabil { get; set; } 


        [ForeignKey("Categorii")]  
        public int IdCategorie { get; set; }  
        public Categorii Categorii { get; set; }  

        [ForeignKey("User")]  
        public int? IdUser { get; set; }  

        public User? User { get; set; }
    }
}
